### PR TITLE
Adapt `FaultDisputeN_1v1_Actors_Test` for nary=2

### PR DESCRIPTION
**Pull request by pochenai**
## Changes
1. rename `moveDirection` to `attackBranch`
2. add `_counterTrace` to solve the DA problem of getting counter's sub claims
3. add DA proof in `_handleStepMove`
4. generalize `AlphabetVM.sol` and `FaultDisputeGame.t.sol` for `SPLIT_DEPTH` and `MAX_GAME_DEPTH`
5. generalize for multi-section fault proof when nary >= 2
6. find a bug in `_findTraceAncestorV2` of `faultDisputeGameN.sol`

## Test
`forge test --mc FaultDisputeN_1v1_Actors_Test  -vvv`

### CODE REVIEW

File: packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol [hunk 1]
```diff
@@ -399,10 +488,16 @@ contract HonestDisputeActor is DisputeActor {
                 assembly {
                     challengeIndex := mload(add(moveData, 0x24))
                 }
+                LibDA.DAItem memory dummyItem = LibDA.DAItem({
+                    daType: LibDA.DA_TYPE_CALLDATA,
+                    dataHash: '00000000000000000000000000000000',
```
Comment (JustXxx) [inline packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol]: dataHash: _dummyClaim().raw(),

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol [hunk 2]
```diff
@@ -2497,6 +2497,8 @@ contract FaultDisputeN_1v1_Actors_Test is FaultDisputeGame_Init {
     DisputeActor internal honest;
     /// @dev The dishonest actor
     DisputeActor internal dishonest;
+    uint256 internal immutable _splitDepth = 2 ** 3;
+    uint256 internal immutable _maxGameDepth = 2 ** 4;
```
Comment (JustXxx) [inline packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol]: Variables that start with an underscore are generally used for function parameters.

File: packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol [hunk 3]
```diff
@@ -40,33 +47,44 @@ abstract contract GameSolver is CommonBase {
...
+        Self,
+        Counter
+    }
+
     /// @notice The `Move` struct represents a move in the game, and contains information
     ///         about the kind of move, the sender of the move, and the calldata to be sent
     ///         to the `FaultDisputeGame` contract by a consumer of this contract.
     struct Move {
         MoveKind kind;
+        uint256 attackBranch;
```
Comment (JustXxx) [inline packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol]: In the FaultDisputeGame contract, we generally use uint64, which is also convenient for use in Go language.

File: packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol [hunk 4]
```diff
@@ -241,36 +251,78 @@ contract HonestGameSolver is GameSolver {
...
+            preStateTrace = absolutePrestateData;
+            preStateItem = LibDA.DAItem({
+                daType: LibDA.DA_TYPE_CALLDATA,
+                dataHash: GAME.absolutePrestate().raw(),
+                proof: hex""
+            });
+        }
+
+        if (_movePos.indexAtDepth() != (1<< (MAX_DEPTH - SPLIT_DEPTH)) - 1 ) {
+             Claim postStateClaim;
```
Comment (JustXxx) [inline packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol]: Unused local variable

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol [hunk 5]
```diff
@@ -2766,32 +2796,37 @@ contract FaultDisputeN_1v1_Actors_Test is FaultDisputeGame_Init {
...
+        uint256 divergeAtOutput = lenOutputs / 2 - 1;
         for (uint256 i; i < dishonestL2Outputs.length; i++) {
-            dishonestL2Outputs[i] = i > 7 ? 0xFF : i + 1;
+            dishonestL2Outputs[i] = i > divergeAtOutput ? 0xFF : i;
         }
-        // The dishonest trace correct is half correct, half incorrect.
-        bytes memory dishonestTrace = new bytes(256);
+        // The dishonest trace is half correct, half incorrect.
+        bytes memory dishonestTrace = new bytes(lenTraces);
+        uint256 divergeAtOffset = 1 << (_maxGameDepth - _splitDepth);
```
Comment (JustXxx) [inline packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol]: Unused local variable

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol [hunk 6]
```diff
@@ -2803,32 +2838,37 @@ contract FaultDisputeN_1v1_Actors_Test is FaultDisputeGame_Init {
...
```
Comment (JustXxx) [inline packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol]: Unused local variable

**Review by JustXxx — APPROVED** [review]

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol (hunk 2 above)
Comment (pochenai) [reply packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol]: Fixed

File: packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol (hunk 3 above)
Comment (pochenai) [reply packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol]: Fixed

File: packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol (hunk 4 above)
Comment (pochenai) [reply packages/contracts-bedrock/test/actors/FaultDisputeActorsN.sol]: Fixed

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol (hunk 5 above)
Comment (pochenai) [reply packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol]: Fixed

File: packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol (hunk 6 above)
Comment (pochenai) [reply packages/contracts-bedrock/test/dispute/FaultDisputeGameN.t.sol]: Fixed

**Pull request merged**